src/phases/move-effect-phase.ts

```diff
@@ -93,6 +93,13 @@ export class MoveEffectPhase extends HitCheckPhase {
     applyMoveAttrs(OverrideMoveEffectAttr, user, targets[0], move, overridden, this.move.virtual);
     // If other effects were overridden, stop this phase before they can be applied
     if (overridden.value) {
+      this.moveHistoryEntry = {
+        moveId: this.move.moveId,
+        targets: this.adjustedTargets ?? this.targets,
+        result: MoveResult.SUCCESS,
+        virtual: this.move.virtual,
+      };
+      user.pushMoveHistory(this.moveHistoryEntry);
```

Comment:
NIT: Consider deconstructing `this.move`:
```suggestion
      const { moveId, virtual } = this.move;
      this.moveHistoryEntry = {
        moveId,
        targets: this.adjustedTargets ?? this.targets,
        result: MoveResult.SUCCESS,
        virtual,
      };
      user.pushMoveHistory(this.moveHistoryEntry);
```

Reply:
Will look into using destructuring for `MoveEffectPhase` separately.